packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -437,60 +457,66 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
 
         // static import or valid string in dynamic import
         // If resolvable, let's resolve it
-        if (specifier) {
+        if (originalUrl) {
           // skip external / data uri
-          if (isExternalUrl(specifier) || isDataUrl(specifier)) {
+          if (isExternalUrl(originalUrl) || isDataUrl(originalUrl)) {
             continue
           }
+
+          // the module graph expects a url without timestamp query
+          let hmrUrl = removeTimestampQuery(stripBase(originalUrl, base))
```

Comment:
Have to remove `?t=` to avoid duplicate module graph entry, but can't overwrite `originalUrl` since we need that to know if the URL was changed.